lib/Doctrine/Common/Lexer/AbstractLexer.php

```diff
@@ -302,6 +304,40 @@ public function getLiteral($token)
         return $token;
     }
 
+    /**
+     * Checks if a given value was caught using a pattern in getCatchablePatterns
+     */
+    public function isCaughtByPattern(?string $value, string $patternKey): bool
+    {
+        if (! isset($this->getCatchablePatterns()[$patternKey]) || ! is_string($value)) {
+            return false;
+        }
+
+        $regex = sprintf('/(%s)/%s', $this->getCatchablePatterns()[$patternKey], $this->getModifiers());
+
+        return (bool) preg_match($regex, $value);
+    }
+
+    /**
+     * Checks if given a value was caught using by any of getCatchablePatterns
+     *
+     * @return string[]
+     */
+    public function caughtByPatterns(string $value): array
+    {
+        $catchables = [];
+
+        foreach ($this->getCatchablePatterns() as $key => $pattern) {
+            if (! $this->isCaughtByPattern($value, $key)) {
```

Comment:
It's weird that you don't get a static analysis warning here… what guarantees that `$key` is going to be a string? It usually is an integer.